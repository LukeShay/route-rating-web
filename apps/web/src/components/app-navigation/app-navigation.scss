:host {
  display: inline;

  #mobile-nav {
    display: none;
  }


  @include mobile {
    display: inline;

    position: fixed;
    bottom: 0;
    width: 100%;

    #mobile-nav {
      display: block;
    }

    #desktop-nav {
      display: none;
    }

    rr-nav-link {
      width: calc(100% / 3);
      float: left;
    }
  }

  rr-nav-bar ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    background-color: $purple;

    rr-nav-link {
      float: left;

      & a {
        display: block;
        color: white;
        text-align: center;
        padding: 14px 16px;
        text-decoration: none;

        &:hover, &.active {
          background-color: darken($purple, 10%);
        }
      }

      &:last-child {
        float: right;
      }
    }
  }
}
