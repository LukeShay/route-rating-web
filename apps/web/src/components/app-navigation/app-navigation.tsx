import { Component, Host, h } from '@stencil/core'

@Component({
  tag: 'app-navigation',
  styleUrl: 'app-navigation.scss',
  shadow: true,
})
export class Navigation {
  render() {
    return (
      <Host>
        <rr-nav-bar>
          <div id="desktop-nav" class="nav-group">
            <rr-nav-link url="/">Home</rr-nav-link>
            <rr-nav-link url="/gyms">Gyms</rr-nav-link>
            <rr-nav-link url="/profile">Profile</rr-nav-link>
          </div>
          <div id="mobile-nav" class="nav-group">
            <rr-nav-link url="/">Home</rr-nav-link>
            <rr-nav-link url="/gyms">Gyms</rr-nav-link>
            <rr-nav-link url="/profile">Profile</rr-nav-link>
          </div>
        </rr-nav-bar>
      </Host>
    )
  }
}
