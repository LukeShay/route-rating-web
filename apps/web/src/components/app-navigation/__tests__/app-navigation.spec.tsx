import { Navigation } from '../app-navigation'
import { newSpecPage } from '@stencil/core/testing'

describe('app-navigation', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Navigation],
      html: `<app-navigation></app-navigation>`,
    })
    expect(page.root).toMatchSnapshot()
  })
})
