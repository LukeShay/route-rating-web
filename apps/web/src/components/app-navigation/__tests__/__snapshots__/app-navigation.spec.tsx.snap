// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-navigation should match snapshot 1`] = `
<app-navigation>
  <mock:shadow-root>
    <rr-nav-bar>
      <div class="nav-group" id="desktop-nav">
        <rr-nav-link url="/">
          Home
        </rr-nav-link>
        <rr-nav-link url="/gyms">
          Gyms
        </rr-nav-link>
        <rr-nav-link url="/profile">
          Profile
        </rr-nav-link>
      </div>
      <div class="nav-group" id="mobile-nav">
        <rr-nav-link url="/">
          Home
        </rr-nav-link>
        <rr-nav-link url="/gyms">
          Gyms
        </rr-nav-link>
        <rr-nav-link url="/profile">
          Profile
        </rr-nav-link>
      </div>
    </rr-nav-bar>
  </mock:shadow-root>
</app-navigation>
`;
